src/ActionList/Item.tsx

```diff
@@ -59,9 +60,50 @@ export interface ItemProps extends React.ComponentPropsWithoutRef<'div'>, SxProp
    * Designates the group that an item belongs to.
    */
   groupId?: string
+
+  /**
+   * Items that are disabled can not be clicked, selected, or navigated through.
+   */
+  disabled?: boolean
+
+  /**
+   * Callback that will trigger both on click selection and keyboard selection.
+   */
+  onAction?: (item: Partial<ItemProps>, event?: MouseEvent | KeyboardEvent) => void
+}
+
+const getItemVariant = (variant = 'default', disabled?: boolean) => {
+  if (disabled) {
+    return {
+      color: themeGet('colors.text.disabled'),
+      iconColor: themeGet('colors.text.disabled'),
+      annotationColor: themeGet('colors.text.disabled'),
+      hoverBackground: 'inherit',
+      hoverCursor: 'default'
+    }
+  }
+
+  switch (variant) {
+    case 'danger':
+      return {
+        color: themeGet('colors.text.danger'),
+        iconColor: themeGet('colors.icon.danger'),
```

Comment:
Non-blocking: I wonder if we can/should have `iconColor` follow `color` in all cases, rather than setting these two color variations explicitly.

The use case I have in mind would be someone passing a `color` system-prop to an `Item`, but I’m not sure whether we want to support that. cc @colebemis, since this is somewhat related to https://github.com/github/primer/issues/115

Reply:
I think it's pretty common to have icon colors differ from the text color. Found an example on this page, in fact:
![image](https://user-images.githubusercontent.com/397836/116438326-5398b700-a81c-11eb-8296-af3e5ab47687.png)

Reply:
Ah, great point and great example—thanks @T-Hugs!

Given “it’s pretty common to have icon colors differ from the text color,” I wonder if we should find a more ergonomic API than what we have now, as demonstrated in Storybook: https://github.com/primer/components/blob/fb77284291c9a0028bd23970f033f60011a3905f/src/stories/ActionList.stories.tsx#L119-L123

`Item` is a container for several children, including the leading visual (e.g. an `Icon`). I’m generally wary of mirroring child props on the parent (e.g. adding an `iconColor` prop to `Icon` that is passed through to as `color` to `Icon`), because:

- It can be ambiguous which child will receive the prop (e.g. would `iconColor` apply to the leading visual, to the trailing visual, or both?)
- It makes a container/parent’s props contract huge/overwhelming (consider the set of all props in `ItemProps` plus all props in `IconProps`)
- Every scenario mirrored-children-props supports can already be addressed with render-props (as in the example above)

That said, maybe my general wariness should be overridden by ergonomic concerns in this specific instance? I’m open to considering any approach.